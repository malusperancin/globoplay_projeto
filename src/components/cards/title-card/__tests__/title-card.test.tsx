/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import { render } from "@testing-library/react";
import TitleCardType from "../../title-card/title-card-type";
import TitleCard from "../title-card";

describe("TitleCard", () => {
  const card: TitleCardType = {
    titleId: "teste",
    headline: "teste",
    description: "teste",
    poster: {
      tv: "teste",
    },
    backgroundUrl: "teste",
    selfRatedContent: true,
    contentRating: "teste",
    type: "teste",
    technicalSpecs: {
      resolutions: [],
    },
    genresNames: [],
    format: "teste",
    releaseYear: 2022,
    contentBrand: {
      trimmedLogo: "teste",
      name: "teste",
    },
    originalContent: true,
    subset: {
      slug: "teste",
    },
  };

  const titleComponent = (
    <TitleCard data={card} isFocus={true} translate={100} />
  );

  const titleComponentNotFocus = (
    <TitleCard data={card} isFocus={false} translate={100} />
  );

  it("should render without throwing an error", () => {
    const { container } = render(titleComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should be focus", () => {
    const { getByTestId } = render(titleComponent);
    const focus =
      getByTestId("title-card").getElementsByClassName("buttonFocus")[0];

    expect(focus).toBeTruthy();
  });

  it("should not be focus", () => {
    const { getByTestId } = render(titleComponentNotFocus);
    const focus = getByTestId("title-card").getElementsByClassName("button")[0];

    expect(focus).toBeTruthy();
  });
});
