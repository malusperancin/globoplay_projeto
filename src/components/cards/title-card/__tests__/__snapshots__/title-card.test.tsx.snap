// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleCard should render without throwing an error 1`] = `
<div>
  <div
    class="container"
    data-testid="title-card"
  >
    <button
      class="buttonFocus"
      style="transform: translate(100px);"
      type="button"
    >
      <img
        alt=""
        class="image"
        src="teste"
      />
    </button>
  </div>
</div>
`;
